scripts/helpers/openai-compatible-agent.py

```
@@ -0,0 +1,98 @@
+    if not key:
+        print(f"ERROR: missing {key_env}", file=sys.stderr); return 2
+    cwd = Path(args.cwd).resolve(); prompt = args.prompt if args.prompt is not None else sys.stdin.read()
+    messages = [
+        {"role":"system","content":"You are a coding agent. Use tools when needed. For implementation tasks, edit files, call git_diff before final, and do not stop after only reading files. Final answer must be visible text. If a verification command fails because a local dependency or tool is missing, stop retrying that same command, call git_diff if not already done, and give a final answer that reports the blocker and the worktree changes."},
+        {"role":"user","content":prompt},
+    ]
+    print(f"provider={args.provider} base_url={base_url} model={model} cwd={cwd}", file=sys.stderr)
+    for turn in range(1, args.max_turns + 1):
+        d = api_call(base_url, key, model, cfg.get("headers", {}), messages)
+        ch = d.get("choices", [{}])[0]; msg = ch.get("message", {})
+        finish = ch.get("finish_reason"); content = msg.get("content") or ""; calls = msg.get("tool_calls") or []
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)